src/components/CreateBlog.vue

```diff
@@ -0,0 +1,209 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+import { useRouter } from 'vue-router';
+
+const props = defineProps<{
+  author: {
+    id: string;
+    name: string;
+    image: string;
+  }
+}>();
+
+
+const category = ref('');
+const title = ref('');
+const description = ref('');
+const image = ref('');
+const router = useRouter();
+
+const handleSubmit = async () => {
+  const blogData = {
+    category: category.value,
+    title: title.value,
+    description: description.value,
+    created_at: Math.floor(Date.now() / 1000), 
+    author: {
+      id: props.author.id,
+      name: props.author.name,
+      image: props.author.image
+    },
+    image: image.value
+  };
+
+  try {
+    const response = await fetch('https://66bc281924da2de7ff69786f.mockapi.io/Blog/', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(blogData),
+    });
```

Comment:
Can you create a composable for this fetch so every time you call it don't need to send the full api endpoint and headers